contracts/Strategy.sol

```diff
@@ -507,16 +494,26 @@ contract Strategy is BaseStrategy, DydxFlashloanBase, ICallee {
         //This part makes sure our withdrawal does not force us into liquidation
         (uint256 depositBalance, uint256 borrowBalance) = getCurrentPosition();
 
-        uint256 AmountNeeded = 0;
-        if(collateralTarget > 0){
-            AmountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
-        }
-        uint256 redeemable = depositBalance.sub(AmountNeeded);
+        uint256 amountNeeded = 0;
+        if(collateralTarget == 0){
+            collateralTarget = 1e15; // 0.001 * 1e18. lower we have issues
+        } 
+        
+        amountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
 
-        if (redeemable < _amount) {
-            cToken.redeemUnderlying(redeemable);
-        } else {
-            cToken.redeemUnderlying(_amount);
+        if(depositBalance >= amountNeeded){
+            uint256 redeemable = depositBalance.sub(amountNeeded);
+
+            if (redeemable < _amount) {
+                cToken.redeemUnderlying(redeemable);
+            
+            } else {
+                cToken.redeemUnderlying(_amount);
```

Comment:
This looks off to me, we are calculating and making decisions based on amountNeeded and other vars, but going into here we use the argument `_amount` that we never check as per docs that must never be over balance, isnt there a chance `_amount` can't be over cToken.BalanceOf??